### DISCUSSION TITLE
请问运行的node之间可以通过add node或者bootnodes相互连接起来吗？

### DISCUSSION THREAD

**Discussion by ruyichang** [nervosnetwork/ckb#2404] [General]
最近发现孤块较多，有没有办法将多地节点间建立连接，减少孤块？谢谢。

**Comment by ruyichang** [comment]
bootnodes列表是最近ckb init --chain mainnet 生成的，发现可能存在区块高度不及时的情况

**Comment by ruyichang** [comment]
whitelist  ?

**Comment by keroro520** [comment]
> whitelist ?

考虑配置 `whitelist_peers` 为自己的对端。节点会优先连接 whitelist 对端。

`whitelist_peers` 的配置格式跟 `bootnodes` 的格式一样，可以通过 RPC [`local_node_info`](https://github.com/nervosnetwork/ckb/blob/develop/rpc/README.md#local_node_info) 拿到目标节点地址。

**Comment by ruyichang** [comment]
> > whitelist ?
> 
> 考虑配置 `whitelist_peers` 为自己的对端。节点会优先连接 whitelist 对端。
> 
> `whitelist_peers` 的配置格式跟 `bootnodes` 的格式一样，可以通过 RPC [`local_node_info`](https://github.com/nervosnetwork/ckb/blob/develop/rpc/README.md#local_node_info) 拿到目标节点地址。

-----------------------------------感谢，请问添加完这些白名单后，初始化的节点，也就是需要同步区块数据的节点优先从白名单同步还是优先bootnodes，因为我4台node互相白名单后发现同步数据较慢而且日志中频繁： Ignoring getheaders from peer=SessionId(887) because node is in initial block download

**Comment by keroro520** [comment]
> 节点优先从白名单同步还是优先bootnodes

优先从白名单连，再连其它节点

> 日志中频繁： Ignoring getheaders from peer=SessionId(887) because node is in initial block download

这说明对端还没同步到最近的高度，仍处于 [IBD 状态](https://bitcoin.org/en/glossary/initial-block-download)  。应该是你填的白名单节点也还没同步到最新？

**Comment by ruyichang** [comment]
> > 节点优先从白名单同步还是优先bootnodes
> 
> 优先从白名单连，再连其它节点
> 
> > 日志中频繁： Ignoring getheaders from peer=SessionId(887) because node is in initial block download
> 
> 这说明对端还没同步到最近的高度，仍处于 [IBD 状态](https://bitcoin.org/en/glossary/initial-block-download) 。应该是你填的白名单节点也还没同步到最新？

----------请问如果我添加白名单后，就只从白名单同步数据吗？感觉添加白名单后同步明显很慢，快24小时了也没同步完成，而且双核机器中一个cpu占满，内存正常

**Comment by keroro520** [comment]
同步慢的话可以考虑更新到 [v0.31.0-pre1](https://github.com/nervosnetwork/ckb/releases/tag/v0.31.0-pre1)（这是预发布版，正式版还没发布），里面有针对同步做了性能优化，应该会快一些。

**Comment by ruyichang** [comment]
我再反馈个问题，还是之前我同时启动4个ckb node，这四个节点互相添加了whitelist_peers，而且whitelist_peers里面也加了其他一些已经完成数据同步的节点，但是这些完成的节点whitelist_peers里面没有加这四台新的node，很奇怪的是只有一台同步完成，剩下3台将近48小时都还差几十万高度同步不完，怀疑是白名单影响了node的同步。

**Comment by ruyichang** [comment]
> 我再反馈个问题，还是之前我同时启动4个ckb node，这四个节点互相添加了whitelist_peers，而且whitelist_peers里面也加了其他一些已经完成数据同步的节点，但是这些完成的节点whitelist_peers里面没有加这四台新的node，很奇怪的是只有一台同步完成，剩下3台将近48小时都还差几十万高度同步不完，怀疑是白名单影响了node的同步。
-----------------------
节点版本ckb:0.29.0 ，换成ckb:0.30.1 也一样的问题
阿里云，essd，2cpu，8G

**Comment by ruyichang** [comment]
可以把白名单的node加到bootnodes吗

**Comment by ruyichang** [comment]
如果设置白名单后whitelist_only默认是true吗？

**Comment by doitian** [comment]
Assigned to @driftluo

**Comment by driftluo** [comment]
> 节点版本ckb:0.29.0 ，换成ckb:0.30.1 也一样的问题
> 阿里云，essd，2cpu，8G

请使用 0.31.x 以上的版本，之前的版本在同步上是有点问题

**Comment by doitian** [comment]
> 如果设置白名单后whitelist_only默认是true吗？

各自独立的配置。如果只开了 whitelist only 没加白名单节点的话就只能 solo 了
